python/pyspark/sql/tests/test_types.py

```diff
@@ -204,6 +204,16 @@ def test_infer_nested_schema(self):
         df = self.spark.createDataFrame(rdd)
         self.assertEqual(Row(field1=1, field2=u'row1'), df.first())
 
+    def test_infer_nested_dict(self):
+        # SPARK-35929: Test inferring nested dict as a struct type.
+        NestedRow = Row("f1", "f2")
+
+        with self.sql_conf({"spark.sql.pyspark.inferNestedDictAsStruct.enabled": True}):
+            nestedRdd = self.sc.parallelize([NestedRow([{"payment": 200.5, "name": "A"}], [1, 2]),
```

Comment:
Can we also test with local dataset instead of RDD? e.g., `spark.createDataFrame([NestedRow([{"payment": 200.5, "name": ", ...)` because the code path is different.